src/server/services/procedures/icesheets/icesheets.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Access to NOAA Global Pliocene-Pleistocene Benthic d18O Stack.
+ *
+ * For more information, check out
+ * https://www.ncdc.noaa.gov/paleo-search/study/5847
+ *
+ * Original datasets are available at
+ * https://www1.ncdc.noaa.gov/pub/data/paleo/contributions_by_author/lisiecki2005/lisiecki2005.txt
+ * @alpha
+ * @service
+ * @category Science
+ * @category Climate
+ */
+const fs = require('fs');
+const path = require('path');
+
+const Icesheets = {};
+
+// this contains delta 18O data, which is a ratio of two oxygen isotopes in deep ocean sediments
+Icesheets._dataDelta18O = [];
+
+// this contains sedimentation rate data, which is a formation rate that is the amount of sediment 
+// deposited during a certain time span, normally expressed as a length per time (e.g., cm/yr).
+Icesheets._dataSedim = [];
+
+// Initialize startyear to -5000000 CE and endyear to 0 CE if no inputs are given.
+let checkInputYear = function(startyear, endyear) {
+    if(startyear == null) {
+        startyear = -5000000;
+    }
+    if(endyear == null) {
+        endyear = 0;
+    }
+    return [startyear, endyear];
+};
+
+// converting time to year CE, note: the present is 1950 here because we are using the BP present
+let timeConversion = function(time) {
+    return (1950 - 1000*time);
+};
+
+// reading and importing delta 18O data
+const importdelta18O = function(line) {
+    let [time, d180, error] = line.trim().split(/\s+/);
+    let adjustTime = timeConversion(time);
```

Comment:
Input validation and casting should be performed here. It will be more performant and should help discover bugs. For example:
```javascript
error = Number(error);
assert(!isNaN(error));  // you will need "const assert = require('assert');" earlier, of course
```

Reply:
Validation for the d18O would be good, too